### DISCUSSION TITLE
k8s alternatives to file storage / PVC

### DISCUSSION THREAD

**Discussion by Ulrar** [Koenkk/zigbee2mqtt#22576] [General]
Hi,

With #21852 it seems like running on k8s is supported, I just wanted to open the discussion on possibly supporting alternative ways of storing data to avoid requiring a PVC, these can be a pain to have highly available on a home cluster.

The obvious method would be kubernetes itself, would there be appetite to support storing the database and any other "dynamic" file in etcd, using CRs ?

You could easily imagine keeping the configuration.yaml file in a configmap, and moving the rest to CRs if the configuration specifies so.

Another could be a simple database backend, although that just moves the pvc requirement around.

**Comment by judahrand** [comment]
I wish that `zigbee2mqtt` supported SQLite as a database backend. If this were the case it would be trivial to set up [litestream](https://github.com/benbjohnson/litestream) to replicate the database to object storage in a sidecar. The database could then be restored on startup with an init container. In my opinion this would be the simplest way to make this work.

Would support for SQLite be welcomed in a PR?
